src/garmin_parser.c

```diff
@@ -564,7 +564,22 @@ DECLARE_FIELD(DEVICE_INFO, firmware, UINT16)
 }
 
 // SPORT
-DECLARE_FIELD(SPORT, sub_sport, ENUM) { garmin->dive.sub_sport = (ENUM) data; }
+DECLARE_FIELD(SPORT, sub_sport, ENUM) {
+	garmin->dive.sub_sport = (ENUM) data;
+	dc_divemode_t val;
+	switch (data) {
+	case 55: val = DC_DIVEMODE_GAUGE;
+		break;
+	case 56:
+	case 57: val = DC_DIVEMODE_FREEDIVE;
+		break;
+	case 63: val = DC_DIVEMODE_CCR;
+		break;
+	default: val = DC_DIVEMODE_OC;
+	}
+	garmin->cache.DIVEMODE = val;
+	garmin->cache.initialized |= 1 << DC_FIELD_DIVEMODE;
+}
```

Comment:
This should use DC_ASSIGN_FIELD(), but it's not wrong as-is.

I have no idea if those case values are right, but I assume you looked them up somewhere and it all looks sane to me.

Reply:
done